### PR TITLE
Fixes #33339 - add routable tabs in host details

### PR DESCRIPTION
**Pull request by amirfefer**
based on https://github.com/theforeman/foreman/pull/8738

### CODE REVIEW

File: webpack/components/AnsibleHostDetail/AnsibleHostDetail.js [hunk 1]
```diff
@@ -1,52 +1,45 @@
...
-const AnsibleHostDetail = ({ response, status }) => {
-  // https://projects.theforeman.org/issues/32398
-  const [activeTab, setActiveTab] = useState('roles');
-  const handleTabClick = (event, tab) => setActiveTab(tab);
+const AnsibleHostDetail = ({
+  response: { id },
+  status,
+  location: { pathname },
+}) => {
+  const routeBuilder = subpath => `#/${ANSIBLE_KEY}/${subpath}`;
```
Comment (xprazak2) [inline webpack/components/AnsibleHostDetail/AnsibleHostDetail.js:16]: nit: same function is in SecondaryTabRoutes, should we extract and reuse it?

**Comment by xprazak2** [comment]
Nice, test failures related though.

File: webpack/components/AnsibleHostDetail/AnsibleHostDetail.js (hunk 1 above)
Comment (amirfefer) [reply webpack/components/AnsibleHostDetail/AnsibleHostDetail.js]: it's not exactly the same, but I extracted these to helper functions

**Comment by xprazak2** [comment]
Linter is complaining now.

**Comment by xprazak2** [comment]
Test failure unrelated, I have seen `TypeError: (0 , _jestGetType.default) is not a function` in other repos as well, but I am not sure why that is happening.

**Comment by xprazak2** [comment]
Thanks!

**Pull request merged**